### PR TITLE
[management-server] Initial changes for Database access

### PR DESCRIPTION
**Pull request by VajiraPrabuddhaka**
## Purpose
This PR brings the initial changes for database access for apk-management-server.

Fixes https://github.com/wso2/apk/issues/120

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,9 @@ target
 # .vscode
 *.vscode
 
+# .idea
+*.idea
+
```
Comment (AmaliMatharaarachchi) [inline .gitignore:43]: there's already an entry for this
```suggestion
```

File: management-server/internal/database/dao.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	var subs []*apkmgt_application.Subscription
+	for rows.Next() {
+		values, err := rows.Values()
+		if err != nil {
+			return nil, err
+		} else {
+			subs = append(subs, &apkmgt_application.Subscription{
+				Uuid:               values[0].(string),
+				ApiUuid:            values[1].(string),
+				PolicyId:           "",
```
Comment (AmaliMatharaarachchi) [inline management-server/internal/database/dao.go:60]: why leave the policy id empty here?

File: management-server/internal/xds/server.go [hunk 3]
```diff
@@ -81,16 +85,175 @@ func FeedData() {
 	newSnapshot, _ := wso2_cache.NewSnapshot(fmt.Sprint(version), map[wso2_resource.Type][]types.Resource{
 		wso2_resource.APKMgtApplicationType: {&applications},
 	})
-	apiCacheMutex.Lock()
-	// TODO (amaliMatharaarachchi) update relevant adapters with snapshot caches
-	apiCache.SetSnapshot(context.Background(), config.ManagementServer.NodeLabels[0], newSnapshot)
-	apiCacheMutex.Unlock()
+	applicationCacheMutex.Lock()
+	defer applicationCacheMutex.Unlock()
+	applicationCache.SetSnapshot(context.Background(), "mine", newSnapshot)
```
Comment (AmaliMatharaarachchi) [inline management-server/internal/xds/server.go:90]:
```suggestion
	applicationCache.SetSnapshot(context.Background(), config.ManagementServer.NodeLabels[0], newSnapshot)
```

File: management-server/internal/xds/server.go [hunk 4]
```diff
@@ -81,16 +85,175 @@ func FeedData() {
...
+		return err
+	}
+	applicationCacheMutex.Lock()
+	defer applicationCacheMutex.Unlock()
+	//performing null check again to avoid race conditions
+	_, errSnap := applicationCache.GetSnapshot(label)
+	if errSnap != nil && strings.Contains(errSnap.Error(), "no snapshot found for node") {
+		errSetSnap := applicationCache.SetSnapshot(context.Background(), label, newSnapshot)
+		if errSetSnap != nil {
+			logger.LoggerXds.Errorf("Error setting empty snapshot to apiCache. error : %v", errSetSnap.Error())
```
Comment (AmaliMatharaarachchi) [inline management-server/internal/xds/server.go:246]: follow proper logs with error codes

File: .gitignore (hunk 1 above)
Comment (VajiraPrabuddhaka) [reply .gitignore]: fixed in https://github.com/wso2/apk/pull/151/commits/4f4c37b97abe8646e783d357a3998601be6de56c

File: management-server/internal/database/dao.go (hunk 2 above)
Comment (VajiraPrabuddhaka) [reply management-server/internal/database/dao.go:60]: Seems policy reference has not yet added to the schema..https://github.com/wso2/carbon-apimgt/blob/apk-refactor/apk/org.wso2.apk.apimgt.impl/src/main/resources/postgresql.sql#L1807-L1824

Will keep a note on this and fix later, when the schema is finalized..

File: management-server/internal/xds/server.go (hunk 3 above)
Comment (VajiraPrabuddhaka) [reply management-server/internal/xds/server.go]: fixed in https://github.com/wso2/apk/pull/151/commits/4f4c37b97abe8646e783d357a3998601be6de56c

File: management-server/internal/xds/server.go (hunk 4 above)
Comment (VajiraPrabuddhaka) [reply management-server/internal/xds/server.go]: fixed in https://github.com/wso2/apk/pull/151/commits/4f4c37b97abe8646e783d357a3998601be6de56c

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**